### PR TITLE
add pyncview and dependencies

### PR DESCRIPTION
**Pull request by jornbr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jornbr** [comment]
@conda-forge/help-python, ready for review!

File: recipes/pyncview/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (ocefpaf) [inline recipes/pyncview/meta.yaml:1]: Ping

Comment (jornbr) [reply recipes/pyncview/meta.yaml]: fixed in latest commit

File: recipes/pyncview/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  imports:
+    - {{ name }}
+  requires:
+    - pip
+  commands:
+    - pip check
+    - multiplot -h
+
+about:
+  home: https://github.com/BoldingBruggeman/{{ name }}
```
Comment (ocefpaf) [inline recipes/pyncview/meta.yaml:43]: We don't abuse the jinja templating b/c it makes the recipe unreadable. We prefer to have the actual name here and in the import section so people can read it and click on links.

File: recipes/pyncview/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+  host:
+    - python >=3.8
+    - pip
+    - hatchling
+  run:
+    - python >=3.8
+    - netcdf4
+    - xmlplot >=0.9.28
+    - matplotlib
```
[the remark this answers is not in the document]
Comment (jornbr) [reply recipes/pyncview/meta.yaml:30]: This change seems harmless, but for reference: the pyncview package depends on pyqt to display its GUI, so changing the matplotlib dependency to matplotlib-base has little benefit (to my understanding, matplotlib = matplotlib-base + pyqt)

File: recipes/pyncview/meta.yaml (hunk 2 above)
Comment (jornbr) [reply recipes/pyncview/meta.yaml]: fixed in latest commit

File: recipes/pyncview/meta.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jornbr) [reply recipes/pyncview/meta.yaml:30]: Given that the knowlegde base states: [Most, if not all, packages that have dependence at runtime on matplotlib should list this dependence as matplotlib-base _unless they explicitly need pyqt_](https://conda-forge.org/docs/maintainer/knowledge_base/#matplotlib) (my emphasis), a final recommendation on whether to make the change above would be welcome. Is there any benefit changing to matplotlib-base, in view of our pyqt dependence?

[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/pyncview/meta.yaml:30]: I missed the qt dependency there, in that case you need full mpl.

**Pull request merged**